templates/lb-fargate-service/cf.yml

```diff
@@ -1,168 +1,174 @@
     Type: AWS::ECS::TaskDefinition
+    DependsOn: LogGroup
     Properties:
+      Family: !Join ['', [!Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName]] # NOTE Do we need to specify this?
+      NetworkMode: awsvpc
+      RequiresCompatibilities:
+        - FARGATE
+      Cpu: !Ref TaskCPU
+      Memory: !Ref TaskMemory
+      ExecutionRoleArn: !Ref ExecutionRole
+      TaskRoleArn: !Ref TaskRole
       ContainerDefinitions:
-      # TODO: support sidecar images
-        - Essential: true
-          # TODO: support images from different repositories
-          Image: !Sub ${Image}
+        - Name: !Ref AppName
+          Image: !Ref ContainerImage
+          PortMappings:
+            - ContainerPort: !Ref ContainerPort {{if .Variables}}
+          # NOTE Can we magically parametrized this?
+          Environment:{{range $name, $value := .Variables}}
+          - Name: {{$name}}
+            Value: {{$value}}{{end}}{{end}}{{if .Secrets}}
+          Secret:{{range $name, $valueFrom := .Screts}}
+          - Name: {{$name}}
+            ValueFrom: {{$valueFrom}}{{end}}{{end}}
           LogConfiguration:
             LogDriver: awslogs
             Options:
-              awslogs-group: !Ref LogGroup
-              awslogs-stream-prefix: Service
               awslogs-region: !Ref AWS::Region
-          Name: web
-          PortMappings:
-            - ContainerPort: 80
-              Protocol: tcp
-      Cpu: 256
-      ExecutionRoleArn: !GetAtt ExecutionRole.Arn
-      Memory: 512
-      NetworkMode: awsvpc
-      RequiresCompatibilities:
-        - FARGATE
-      TaskRoleArn: !GetAtt TaskRole.Arn
-
-  LogGroup:
-    Type: AWS::Logs::LogGroup
-    UpdateReplacePolicy: Retain
-    DeletionPolicy: Retain
-
+              awslogs-group: !Ref LogGroup
+              awslogs-stream-prefix: ecs
   ExecutionRole:
     Type: AWS::IAM::Role
     Properties:
       AssumeRolePolicyDocument:
         Statement:
-          - Action: sts:AssumeRole
-            Effect: Allow
+          - Effect: Allow
             Principal:
               Service: ecs-tasks.amazonaws.com
-        Version: 2012-10-17
-
-  LogWritingPolicy:
-    Type: AWS::IAM::Policy
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
+  TaskRole:
+    Type: AWS::IAM::Role
     Properties:
-      PolicyDocument:
+      AssumeRolePolicyDocument:
         Statement:
-          - Action:
-              - logs:CreateLogStream
-              - logs:PutLogEvents
-            Effect: Allow
-            Resource: !GetAtt LogGroup.Arn
-        Version: 2012-10-17
-      PolicyName: LogWritingPolicy
-      Roles: [ !Ref ExecutionRole ]
-
+          - Effect: Allow
+            Principal:
+              Service: ecs-tasks.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/AdministratorAccess' # NOTE Are we sure about this??
```

Comment:
Nope! But can we use PowerUser instead?

Reply:
Does PowerUser have the permissions that we want?
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "NotAction": [
                "iam:*",
                "organizations:*",
                "account:*"
            ],
            "Resource": "*"
        },
        {
            "Effect": "Allow",
            "Action": [
                "iam:CreateServiceLinkedRole",
                "iam:DeleteServiceLinkedRole",
                "iam:ListRoles",
                "organizations:DescribeOrganization",
                "account:ListRegions"
            ],
            "Resource": "*"
        }
    ]
}
```

Reply:
Yea - the first statement lets it do everything *except* what's listed. It's essentially an Admin role that can't create arbitrary roles.

Reply:
ooh I missed that it's a negation **Not**Action 🤦‍♂️

Reply:
Done! thanks 👍